solver/src/main/java/org/chocosolver/solver/constraints/real/RealPowerPropagator.java

```diff
@@ -0,0 +1,108 @@
+package org.chocosolver.solver.constraints.real;
+
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.constraints.PropagatorPriority;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.expression.continuous.arithmetic.RealIntervalConstant;
+import org.chocosolver.solver.variables.RealVar;
+import org.chocosolver.solver.variables.events.RealEventType;
+import org.chocosolver.util.ESat;
+import org.chocosolver.util.objects.RealInterval;
+import org.chocosolver.util.tools.RealUtils;
+
+/**
+ * It's a pure java alternative to ibex that must be used to calculate
+ * power operations (y = x^k) with integer positive exponents (k).
+ *
+ * <p>
+ * Just in case, this is an external library for interval arithmetic.
+ * Reference: https://jitpack.io/p/jinterval/jinterval#readme
+ * <p>
+ * This implementation is based from the chapter 16.1 "From Discrete to
+ * Continuous Constraints", example 16.2 [1].
+ * <p>
+ * [1] Francesca Rossi, Peter van Beek, and Toby Walsh. 2006. Handbook of
+ * Constraint Programming. Elsevier Science Inc., USA.
+ *
+ * @author João Pedro Schmitt
+ * @since 08/09/2020
+ */
+public class RealPowerPropagator extends Propagator<RealVar> {
+
+    private RealVar yVar;
+
+    private RealVar xVar;
+
+    private int k;
+
+    public RealPowerPropagator(RealVar yVar, RealVar xVar, int k) {
+        super(new RealVar[]{yVar, xVar}, PropagatorPriority.BINARY, false);
+        this.yVar = vars[0];
+        this.xVar = vars[1];
+        this.k = k;
+    }
+
+    @Override
+    public int getPropagationConditions(int vIdx) {
+        return RealEventType.BOUND.getMask();
+    }
+
+    @Override
+    public ESat isEntailed() {
+        ESat result = ESat.UNDEFINED;
+        if (isCompletelyInstantiated()) {
+            double lb = yVar.getLB() - Math.pow(xVar.getLB(), k);
+            double ub = yVar.getUB() - Math.pow(xVar.getUB(), k);
+            lb = RealUtils.roundValue(lb, xVar.getPrecision());
+            ub = RealUtils.roundValue(ub, xVar.getPrecision());
+            result = ESat.eval(lb <= 0 && 0 <= ub);
+        }
+        return result;
+    }
+
+    @Override
+    public final void propagate(int evtmask) throws ContradictionException {
+        boolean hasChanged = true;
+        // Projection + propagation
+        while (hasChanged) {
+            hasChanged = kPower();
+            hasChanged |= kRoot();
+        }
+    }
+
+    private boolean kPower() throws ContradictionException {
+        RealInterval x = new RealIntervalConstant(xVar.getLB(), xVar.getUB());

```

Comment:
`xVar` extends `RealInterval`, creating a `RealIntervalConstant ` is not necessary.